core/Plugin/Visualization.php

```diff
@@ -461,10 +461,30 @@ private function postDataTableLoadedFromAPI()
         if ($this->dataTable instanceof DataTable) {
             $this->metadata = $this->dataTable->getAllTableMetadata();
 
-            if (isset($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
-                $this->reportLastUpdatedMessage = $this->makePrettyArchivedOnText();
+
+        } else {
+            // if the dataTable is Map
+            if ($this->dataTable instanceof DataTable\Map) {
+                // load all the data
+                $dataTable = $this->dataTable->getDataTables();
+                // find the latest key
+                foreach ($dataTable as $item) {
+                    $metaData = $item->getAllTableMetadata();
+                    // if ts_created not exist exit loop
+                    if (empty($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        continue;
+                    }
+                    // if ts_created not record in the metadata or if current is more recent up the metadata
+                    if (!empty($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])
+                      || strtotime($metaData[DataTable::ARCHIVED_DATE_METADATA_NAME]) > strtotime($this->metadata[DataTable::ARCHIVED_DATE_METADATA_NAME])) {
+                        $this->metadata = $metaData;
```

Comment:
`$this->metadata` could now be converted to a local variable in the method only. As the property is only used in that one method, there is no use in holding it as private class property.